python/cudf/cudf/core/column/column.py

```diff
@@ -2001,9 +2000,6 @@ def as_column(
                 f"{type(arbitrary).__name__} with "
                 f"{type(arbitrary.dtype).__name__} is not supported."
             )
-        if dtype is not None:
```

Comment:
Why did we lose this cast?

Reply:
Now that all the branches in this parent "pandas object branch" `return`s now, we no longer need to potentially `astype` since the `dtype` is passed along